### ISSUE TITLE
JavadocStyle does not properly detect incomplete/missing/unclosed HTML end tags in Javadoc

### ISSUE THREAD

**Issue by dhendriks** [checkstyle/checkstyle#1112]
The 'JavadocStyle' check, according to the documentation, should check for:

```
Check text for incomplete HTML tags. Verifies that HTML tags have
corresponding end tags and issues an "Unclosed HTML tag found:"
error if not. An "Extra HTML tag found:" error is issued if an end tag
is found without a previous open tag. 
```

The 'checkHtml' property ("Whether to check for incomplete HTML tags.") is a 'boolean' property with default value 'true', so using default settings, it should check for this.

However, for the following Java file:

```java
/** Text. */
public class Test {
    /**
     * Some text.
     *
     * <p>
     *   <ul>
     *     <li>Proper item, with closing 'li' tag.</li>
     *     <li>First 'li' to not have a closing tag.
     *     <li>Second 'li' to not have a closing tag.
     *     <li>Proper item, with closing 'li' tag.</li>
     *   </ul>
     * </p>
     */
    public void a() {
        // ...
    }

    /**
     * Some text.
     *
     * <p>Missing closing 'p' tag.
     */
    public void b() {
        // ...
    }
}
```

No problems are reported. I would expect 3 problems to be reported: two for missing 'li' closing tags (for method 'a'), and one for a missing 'p' closing tag (for method 'b').

To me, this looks like a bug.

**Comment by romani** [comment]
current javadoc implementation is extremely buggy, i think we will find time to reimplement the Check base on javadoc parser that already implemented. Feel fre to contribute.

**Comment by pbludov** [comment]
The reality is slightly different:
```
$ javadoc Test.java
Loading source file Test.java...
Constructing Javadoc information...
Standard Doclet version 1.8.0_151
Building tree for all the packages and classes...
Generating ./Test.html...
Test.java:6: warning: empty <p> tag
     * <p>
       ^
Test.java:13: error: unexpected end tag: </p>
     * </p>
       ^
1 error
1 warning

Everything is ok with method b() and the list items.
But closing </p> for method a() is misplaced and treated as an error by javadoc tool.
```

**Comment by pbludov** [comment]
The javadoc tool is uses an ancient HTML4 parser and stuck with it for compatibility reasons.
Perhaps this will change in future versions, but I doubt it. At least, JDK 9 still complain about unexpected closing paragraph tag.

**Comment by romani** [comment]
we should not mimic javadoc tool problems, one day it will be fixed in javadoc tool.

**Comment by pbludov** [comment]
>one day it will be fixed in javadoc tool.
Seems to be fixed in JDK9. The javadoc tool from JDK9 has option -html5 to generate HTML5 output.

**Comment by romani** [comment]
JavadocStyle is removed from Checkstyle at 13.9.0, users should use SummaryJavadoc instead
Details:  https://github.com/checkstyle/checkstyle/issues/20610
